### DISCUSSION TITLE
How to customize the internal parameter of nodes?

### DISCUSSION THREAD

**Discussion by FreshLucas-git** [bcakmakoglu/vue-flow#727] [Q&A] [answered]
Thank you for your greart work. This library is very helpful. I want to pass the argument or state into custom node as internal parameters.
After that, I want to log the node's state, including type, position, label, and custom parameters. Maybe Pinia is the solution for this problem. Is there any way to implement it?

**Comment by bcakmakoglu** [comment, marked as answer]
If I understand you correctly, you want to pass some additional info to your nodes.
You can use `node.data` for this purpose

```js
{
   id: '1',
   label: 'Node 1',
   position: { x: 0, y: 0 },
   data: {
      additionalInfo: 'foobar'
   }
}
```

Accessing the values can be done multiple ways, really depends from *where* you want to access it.

From inside a custom node via props
```vue
// CustomNode.vue
<script setup>
const props = defineProps(['id', 'type', 'label', 'data'])

console.log(props)
</script>
```

Or from inside a custom node using `useNode`
```vue
// CustomNode.vue
<script setup>
const { node } = useNode()

console.log(node)
</script>
```

Or, maybe you want to access the node from some Sidebar component
```vue
// Sidebar.vue
<script setup>
const { findNode } = useVueFlow()

const node = findNode(someId)

console.log(node)
</script>
```

**Reply by FreshLucas-git** [reply]
Thank you very much for your kindly explanation. You're amazing. :smile:
